src/test/resources/test-scripts/late_binding.bsh

```diff
@@ -0,0 +1,22 @@
+#!/bin/java bsh.Interpreter
+
+source("TestHarness.bsh");
+source("Assert.bsh");
+
+/** Late binding or namespace chaining support as per #676  */
+int bar = 42;
+int getBar() { return bar; }
+{
+   {
+      int bar = 4711;
+      {
+         int bar = 5;
+         assertEquals("Bind method to new declaration value", 5, getBar());
+      }
+      assertEquals("Bind method to previous declared value", 4711, getBar());
+   }
+   assertEquals("Bind method to original declared value from parent scope", 42, getBar());
+}
+assertEquals("Bind method to declared variable from mehod declared scope", 42, getBar());
+
```

Comment:
> Here is java code that you can use as the basis for a unit test for this problem. When this is run through BeanShell the asserts should not trigger.

This does not appear to run in BeanShell so how do we know it will fail? It is not a function but a class, lexically anchored by the Internal instance. The scripted example does not have any of those restrictions. They have lexical scope from point of execution all the way to global.